modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/rulesSupport.js

```diff
@@ -373,11 +373,22 @@ export const isActionsValid = (actions) =>
 				case 'auto-fill': {
 					const {inputs, outputs} = payload;
 
-					return (
-						Boolean(target) &&
-						(Boolean(Object.values(inputs).length) ||
-							Boolean(Object.values(outputs).length))
+					let valid =
+						Boolean(Object.values(inputs).length) ||
+						Boolean(Object.values(outputs).length);
+
+					const currentDataProvider = dataProviders.filter(
+						(provider) => provider.id === payload.label
 					);
```

Comment:
Take advantage of the destructuring:
https://github.com/liferay-forms/liferay-portal/blob/86d9bdb0cf5ead13a3cdd9b1b05495145cf9b696/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/rulesSupport.js#L374
```javascript
const {inputs, label, outputs} = payload;
``` 

```suggestion
					const currentDataProvider = dataProviders.filter(({id}) => id === label);
```